src/main/java/me/steinborn/krypton/mixin/shared/network/flushconsolidation/ClientConnectionMixin.java

```diff
@@ -44,7 +43,8 @@ private void initAddedFields(CallbackInfo ci) {
             cancellable = true,
             method = "sendImmediately",
             at = @At(value = "FIELD", target = "Lnet/minecraft/network/ClientConnection;packetsSentCounter:I", opcode = Opcodes.PUTFIELD, shift = At.Shift.AFTER))
-    private void sendImmediately$rewrite(Packet<?> packet, @Nullable GenericFutureListener<? extends Future<? super Void>> callback, CallbackInfo info, NetworkState packetState, NetworkState protocolState) {
+    private void sendImmediately$rewrite(Packet<?> packet, @Nullable class_7648 callback, CallbackInfo info, NetworkState packetState, NetworkState protocolState) {
```

Comment:
Very interesting that Mojang would introduce this particular abstraction, given that it could have been easily accomplished via the future callback.